src/lmql/models/lmtp/lmtp_replicate_client.py

```diff
@@ -0,0 +1,144 @@
+            return
+
+        # We have identifier and version; just need to make sure they actually exist
+        async with self.session.get(f'https://api.replicate.com/v1/models/${self.model_identifier}/versions/${self.model_version}',
+                    headers={
+                        'Authorization': f'Token {self.api_key}',
+                        'Content-Type': 'application/json'
+                    },
+            ) as resp:
+                if resp.status != 200:
+                    explanation = await resp.text()
+                    raise Exception(f'Unhandled error: model lookup of {self.model_identifier} version ${self.model_version} failed with status {resp.status}: {explanation}')
+
+        self.model_validated = True
+
+
+    async def submit_batch(self, batch):
+        if self.model_version is None or not self.model_validated:
+            await self.check_model()
+        # FIXME: Maybe store id to use for later cancel calls?
+        body = {"input": {"ops_batch_json": json.dumps(batch)}, "stream": True, "version": self.model_version}
+        async with self.session.post('https://api.replicate.com/v1/predictions',
+                headers={
+                    'Authorization': f'Token {self.api_key}',
+                    'Content-Type': 'application/json'
+                }, json=body) as resp:
+            if resp.status != 201:
+                explanation = await resp.text()
+                raise Exception(f'Unhandled error: prediction creation failed with status {resp.status}: {explanation}')
+            creation_result = await resp.json()
+            stream_url = creation_result['urls']['stream'] # note there's also a cancel url provided
+        async with aiohttp_sse_client.client.EventSource(stream_url,
+                session=self.session,
+                headers={
+                    'Accept': 'text/event-stream',
+                    'Authorization': f'Token {self.api_key}',
+                }) as event_source:
+            async for event in event_source:
+                if event.type == 'done':
+                    return
+                content = event.data.rstrip('\n').split('\n')
+                for line in content:
+                    if len(line) == 0:
+                        continue
+                    if not ' ' in line:
+                        warnings.warn(f'Misformatted content received from Replicate task: {line!r}')
+                        continue
+                    cmd, item_json = line.split(' ', 1)
+                    if cmd != 'TOKEN':
+                        warnings.warn(f"Unknown command {cmd!r} in line {line!r} -- protocol version mismatch? {cmd!r}")
+                        continue
+                    item = json.loads(item_json)
+                    if item.get("error") is not None:
+                        # FIXME: Is it appropriate for this to kill the whole batch, not just the single stream?
+                        raise LMTPStreamError(item["error"])
```

Comment:
Very likely this is appropriate, since server-side an exception during generate also will take down the entire batch, see https://github.com/eth-sri/lmql/blob/main/src/lmql/models/lmtp/lmtp_scheduler.py#L279. 

In the future we could try to handle this, but for now it should be fine.